File: strategy/marvin/agent/shared/movecommand.lua
local Base = require "agent/base/behavior"
local MoveCommand = Class("Agent.Shared.MoveCommand", Base)

local World = require "../base/world"
local MoveToPos = require "task/movetopos"


function MoveCommand:check()
	return self._robot.moveCommand ~= nil and not World.IsSimulated
end

function MoveCommand:_updateTask()
	return MoveToPos, {self._robot.moveCommand, nil, nil, nil, true}, true
end

return MoveCommand

File: strategy/typescript/glados/agent/shared/movecommand.ts
import * as pb from "base/protobuf";
import * as World from "base/world";

import { Behavior, TaskAssignment } from "glados/agent/base/behavior";
import { MoveToPos } from "glados/task/shared/movetopos";

export class MoveCommand extends Behavior {
	public check(): Behavior | undefined {
		return this._robot.moveCommand != undefined && World.WorldStateSource() !== pb.world.WorldSource.INTERNAL_SIMULATION
			? this
			: undefined;
	}

	protected _updateTask(): TaskAssignment<typeof MoveToPos> {
		return [MoveToPos, [{ pos: this._robot.moveCommand!.pos, ignoreDefaultObstacles: true }], true];
	}
}
